src/map/status.cpp

```diff
@@ -1000,8 +1000,15 @@ int status_damage(struct block_list *src,struct block_list *target,int64 dhp, in
 			for (const auto &it : status_db) {
 				sc_type type = static_cast<sc_type>(it.first);
 
-				if (sc->data[type] && it.second->flag[SCF_REMOVEONDAMAGED])
+				if (sc->data[type] && it.second->flag[SCF_REMOVEONDAMAGED]) {
+					if (sc->opt1 > OPT1_NONE) {
+						unit_data *ud = unit_bl2ud(target);
```

Comment:
Considering you have so much work with storing this in ud and then also loading it back (because in both places you can't access ud directly), wouldn't it be better to store it inside sc?